PowerEditor/src/ScintillaComponent/AutoCompletion.cpp

```diff
@@ -183,12 +182,6 @@ bool AutoCompletion::showApiAndWordComplete()
 		{
 			if (!isInList(_keyWordArray[i], wordArray))
 				wordArray.push_back(_keyWordArray[i]);
-			canStop = true;
-		}
-		else if (canStop)
-		{
-			// Early out since no more strings will match
-			break;

```

Comment:
Why `canStop` is removed?

Reply:
I used `OutputDebugString()` in the `for` loop. This is the saved log from Sysinternals DebugView from testing:

```
00000001	7:17:35 PM	[4012] Tra	
00000002	7:17:35 PM	[4012] int: "1"	
00000003	7:17:35 PM	[4012] TRA	
00000004	7:17:35 PM	[4012] TRA	
00000005	7:17:35 PM	[4012] TRA	
00000006	7:17:35 PM	[4012] TRA	
00000007	7:17:35 PM	[4012] TRA	
00000008	7:17:35 PM	[4012] TRA	
00000009	7:17:35 PM	[4012] TRA	
00000010	7:17:35 PM	[4012] TRA	
00000011	7:17:35 PM	[4012] TRA	
00000012	7:17:35 PM	[4012] TRA	
00000013	7:17:35 PM	[4012] TRA	
00000014	7:17:35 PM	[4012] TRA	
00000015	7:17:35 PM	[4012] -- break	
00000016	7:17:35 PM	[4012] -- break	
00000017	7:17:35 PM	[4012] -- break	
00000018	7:17:35 PM	[4012] -- break	
00000019	7:17:35 PM	[4012] -- break	
00000020	7:17:35 PM	[4012] -- break	
00000021	7:17:35 PM	[4012] -- break	
00000022	7:17:35 PM	[4012] -- break	
00000023	7:17:35 PM	[4012] -- break	
00000024	7:17:35 PM	[4012] -- break	
00000025	7:17:35 PM	[4012] -- break	
00000026	7:17:35 PM	[4012] Tra	
00000027	7:17:35 PM	[4012] Tra	
00000028	7:17:35 PM	[4012] -- break	
00000029	7:17:35 PM	[4012] -- break	
00000030	7:17:35 PM	[4012] -- break	
```

up to `00000462`.

 * 1st line is value of `beginChars`, which is what was typed into the edit pane.
 * 2nd line is value of `_ignoreCase`, which is to ignore case as shown.
 * 3rd to 462nd is the values of `kwSufix.c_str()` and the `break` statement is commented so that ` -- break` is printed out instead.

With `canStop` set to `true` on the 1st break, then only the `TRA` entries are processed and not the `Tra` entries. This is probably the main issue causing the missing entries in the autocomplete list.

`canStop` removed as it does not allow all entries for the autocomplete list.

Reply:
Just tested your PR:
It seems work with [test.xml](https://subversion.assembla.com/svn/omnibot/Enemy-Territory/0.8/tools/Notepad++/autoCompletion/omni-bot.xml). However, if we change ignoreCase to "no" in test.xml:
`<Environment ignoreCase="no" startFunc="(" stopFunc=")" paramSeparator="," terminal=";" />`
It doesn't work at all - the list contains a lot entries which should be excluded.

Reply:
At set on `ignoreCase` set to `no` in *test.xml*.

I get the same lists as in the issue report. `Tra` shows 2 entries like as the 7.3.3 list and `TRA` shows 12 entries like as the 8.3 list.

I can only recommend retest to see if same.

Reply:
Also make sure the setting **Preferences -> Auto-completion -> From `3`th character** is set so the autocomplete list is based on the 3rd character, not the 1st character. The autocomplete list should be more filtered on the 3rd character.